contract/src/main/kotlin/com/ritense/valtimo/contract/case_/CaseDefinitionId.kt

```diff
@@ -44,6 +44,8 @@ data class CaseDefinitionId(
             ?: throw IllegalArgumentException("Given version '$versionTag' is not a valid Semver version")
     )
 
+    constructor(caseDefinitionId: String) : this(caseDefinitionId.split(':')[0], caseDefinitionId.split(':')[2])
```

Comment:
We might want to add a check to see if the array length is 2 so we can filter out some of the garbage input.